bsve/server/bsve_wms.py

```diff
@@ -103,3 +124,26 @@ def createBsveDataset(self, params, layer_type):
     createBsveSource.description = (
         Description('Create bsve datasets from bsve geoserver')
     )
+
+    def getLayers(self, user):
+        folder = findDatasetFolder(
+            user, user
+        )
+        if not folder:
+            return []
+
+        items = self.model('folder').childItems(folder)
+
+        layers = {}
+        for item in items:
+            adapter = item.get('meta', {}).get('minerva', {}).get('adapter')
+            name = item.get('meta', {}).get('minerva', {}).get('type_name')
+            if adapter == 'bsve':
+
+                # delete duplicates if they exist
+                if name in layers:
+                    self.model('item').remove(item)
```

Comment:
Because the name being compared here is the unique layer id used by the geoserver instance, this shouldn't actually happen if everything works correctly.  I added it to handle layers that were added before this code was in place.  Having two entries pointing to the same layer would break the fact that the layers in geoviz are effectively a cache on top of the geoserver instance.  The most sensible thing to do when confronted invalid data in a cache is to delete it and refresh from the source.